### PR TITLE
fix #1049: fix potential nil dereference

### LINKED ISSUE #1049: potential nil dereference
**Describe the bug**

```
{"log":"[APP]time=\"2019-05-06T12:22:46Z\" level=error msg=\"error while processing filters during response: apiUsageMonitoring: runtime error: invalid memory address or nil pointer dereference\"
","stream":"stderr","time":"2019-05-06T12:22:46.944219058Z"}
```

**To Reproduce**

Currently not possible. If the fix for this issue does not solve the problem, we'll have to wait for #1045 to find the root cause.

### PR DESCRIPTION
**Pull request by zeitlinger**
Closes #1049.

### CODE REVIEW

**Comment by tkrop** [comment]
:+1:

File: filters/apiusagemonitoring/filter.go [hunk 1]
```diff
@@ -133,19 +134,22 @@ func (f *apiUsageMonitoringFilter) getRealmClientKey(r *http.Request, path *path
...
 	return realm + "." + client
 }
 
-// resolvePath tries to match the request's path with one of the configured path template.
-func (f *apiUsageMonitoringFilter) resolvePath(req *http.Request) *pathInfo {
-	for _, p := range f.Paths {
-		if p.Matcher.MatchString(req.URL.Path) {
-			return p
+// resolveMatchedPath tries to match the request's path with one of the configured path template.
+func (f *apiUsageMonitoringFilter) resolveMatchedPath(url *url.URL) *pathInfo {
```
Comment (szuecs) [inline filters/apiusagemonitoring/filter.go]: the var name `url` shadows the url stdlib package. Can you please change the var to `u` for example?

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by szuecs** [comment]
:-1:

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**